kvstore/tokenrenewer.go

```diff
@@ -0,0 +1,57 @@
+package kvstore
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	"github.com/hashicorp/vault/api"
+)
+
+// TokenRenewer - struct responsible for Vault token renewal process
+type TokenRenewer struct {
+	client               *api.Client
+	tokenRenewalInterval time.Duration
+	tokenTTL             time.Duration
+}
+
+// NewTokenRenewer - returns a new token renewer
+func NewTokenRenewer(client *api.Client, tokenRenewalInterval, tokenTTL time.Duration) (*TokenRenewer, error) {
```

Comment:
I don't see this being called anywhere in the rest of the code. Does this mean we have to call it manually?

Reply:
Yes, it needs to be called separately from the client's code and you'd probably want to run it only for periodic tokens. 
Here's the draft example of how to use it in Release Agent: https://github.com/magneticio/vamp-release-agent/pull/127

Reply:
Wouldn't it be better to call it before any operation in the client itself instead of having to modify every project that uses this library to do that?

Reply:
There are two problems with that:
1. If the client does not perform any action in Vault for a longer period, the token can expire after reaching the TTL.
2. From the performance perspective, it will add one additional call for every single Vault request.

Reply:
And still you'd need to modify every project that uses this library, because you'd need to pass token TTL and renewal interval

Reply:
So you want to have some kind of routine in every app that epriodically renews the token? 
In that case wouldn't it make sense to have this routine be part of the client? Since most apps will have a single client anyway.

Regarding your points:
1) true, but if the client is used by the realease agent that will never happen. Other component can have a much longer TTL so yeah, if you don't use them for one year the token expires, but that's logical.
2) I'm not sure it's really one extra call every time, since you can renew only when half of the ttl is past.
3) Can't the token ttl be retrieved? and the renewal interval can be based on the ttl. Maybe I'm wrong however I didn't check thoroughly